File: app/src/main/java/com/photoeditor/sketch/photo/maker/editor/sketchphotoeditor/pencilsketching/sketchphoto/Ragnarok/OilFilter.java
package com.photoeditor.sketch.photo.maker.editor.sketchphotoeditor.pencilsketching.sketchphoto.Ragnarok;

import android.graphics.Bitmap;

class OilFilter {
    static {
        System.loadLibrary("AndroidImageFilter");
    }

    static Bitmap changeToOil(Bitmap bitmap, int oilRange) {
        int width = bitmap.getWidth();
        int height = bitmap.getHeight();

        int[] pixels = new int[width * height];
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height);

        int[] returnPixels = NativeFilterFunc.oilFilter(pixels, width, height, oilRange);
        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888);
    }
}

File: app/src/main/java/wishpool/sketch/ragnarok/OilFilter.kt
package wishpool.sketch.ragnarok

import android.graphics.Bitmap

internal object OilFilter {
    init {
        System.loadLibrary("AndroidImageFilter")
    }

    fun changeToOil(bitmap: Bitmap, oilRange: Int): Bitmap {
        val width = bitmap.width
        val height = bitmap.height
        val pixels = IntArray(width * height)
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height)
        val returnPixels = wishpool.sketch.ragnarok.NativeFilterFunc.oilFilter(pixels, width, height, oilRange)
        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888)
    }
}